docs/cloud/dev/spryker-cloud-commerce-os/configuring-deployment-pipelines/deployment-in-states.md

```diff
@@ -0,0 +1,132 @@
+description: Deployment pipelines consist of three configurable stages.
+template: howto-guide-template
+originalLink: https://cloud.spryker.com/docs/deployment-pipelines
+originalArticleId: 14d91c9f-6c4e-4481-83ee-005683ce602f
+redirect_from:
+  - /docs/deployment-pipelines
+  - /docs/en/deployment-pipelines
+  - /docs/cloud/dev/spryker-cloud-commerce-os/deployment-pipelines/deployment-pipelines.html
+---
+
+When it comes to complex application, deploying to production environments is not just going from version one to version two. This document describes the states though which an application goes through during a deployment and how they affect its behavior.
+
+## Prerequisites
+
+To learn how pipelines work in SCCOS, see [Deployment pipelines](/docs/cloud/dev/spryker-cloud-commerce-os/configuring-deployment-pipelines/deployment-pipelines.html).
+
+## Production Pipeline
+
+Here is the list of the steps that we will find in a regular pipeline. Highlighted in **bold** are the steps that have a potential to break (or at least impact) our application.
+* Source
+* Please_approve
+* **Build_and_Prepare**
+* **Configure_RabbitMQ_Vhosts_and_Permissions**
+* **Run_pre-deploy_hook**
+* Pre_Deployment_Configuration
+* **Deploy_Scheduler**
+* **Run_install**
+* **Deploy_Spryker_services**
+* UpdateDeployedVersion
+* Post_Deployment_Configuration
+* **Run_post-deploy_hook**
+
+Let's see one by one how they can affect our application. Please keep in mind that this is a pessimistic approach, in order to show the potential issues within our deployments.
+It does not reflect the reality of a pipeline deployment.
+
+### Initial state
+This is how our application behaves  when no pipeline is running. In a fully working state:
+
+![Initial state](./images/initial_state/initial-state.gif)
+
+### Build_and_Prepare
+
+In this step, we build the Containers of each service that we are going to deploy (zed, yves, frontend, backoffice, backapi, glue, etc.)
+For the sake of simplicity, we will use only Glue and Zed in our examples. 
+
+![Build_and_Prepare](./images/Build_and_Prepare/Build_and_Prepare.jpg)
+
+### Configure_RabbitMQ_Vhosts_and_Permissions
+
+Here we update Rabbit MQ vhosts, users and permissions. 
+It should rarely be updated, but in this pessimistic approach, we want to illustrate what would happen if a request would come in while  RabbitMQ is updated:
+
+![Configure_RabbitMQ_Vhosts_and_Permissions](./images/Configure_RabbitMQ_Vhosts_and_Permissions/rmq.gif)
+
+### Run_pre-deploy_hook
+
+This step will run our application defined scripts. They are registered under the constant `SPRYKER_HOOK_BEFORE_DEPLOY` of our deploy.yml file. By default it  runs: 
+```shell
+vendor/bin/install -r pre-deploy -vvv
+```
+We mention this step here because it is important to understand that if our script takes a long time to run, we need to keep in mind all the services that are currently (or soon) in an updated state might respond incorrectly to requests. 
```

Comment:
Do you mean the services that are being updated by the scripts in this step?

Reply:
Everything that is being transitioned (in yellow) interacting with what hasn't (in white)

Reply:
there is no diagram for it, so it's hard to imagine what can be affected